src/ebmc/k_induction.cpp

```diff
@@ -94,6 +95,8 @@ int k_inductiont::operator()()
     return 1;
   }
 
+  solver_factory = ebmc_solver_factory(cmdline);
```

Comment:
It's a bit weird that this member is only (re-)initialised here. I started wondering whether it should be a pointer instead.

Reply:
The container is effectively a pointer, so I don't think there's any performance problem here. It might be prettier to do the initialisation in the constructor.